nifi-external/nifi-spark-receiver/src/main/java/org/apache/nifi/spark/NiFiReceiver.java

```diff
@@ -162,17 +162,19 @@ public void run() {
                         }
 
                         final List<NiFiDataPacket> dataPackets = new ArrayList<>();
+                        InputStream inStream =null;
                         do {
                             // Read the data into a byte array and wrap it along with the attributes
                             // into a NiFiDataPacket.
-                            final InputStream inStream = dataPacket.getData();
+                            inStream = dataPacket.getData();
                             final byte[] data = new byte[(int) dataPacket.getSize()];
                             StreamUtils.fillBuffer(inStream, data);
 
                             final Map<String, String> attributes = dataPacket.getAttributes();
                             final NiFiDataPacket NiFiDataPacket = new StandardNiFiDataPacket(data, attributes);
                             dataPackets.add(NiFiDataPacket);
                             dataPacket = transaction.receive();
+                            inStream.close();
```

Comment:
While generally it is indeed necessary to close the resources to avoid unnecessary memory leaks, this is not the right place to do that since NiFiReceiver is not the creator/owner of the stream and if another handle exist for such stream it could lead to unexpected results. We should probably investigate where this stream is created and ensure that it is closed there.